sky/resources.py

```diff
@@ -1389,6 +1389,11 @@ def __setstate__(self, state):
         if version < 2:
             self._region = None
 
+        # spot_recovery is deprecated. We do not need to set the _spot_recovery
+        # here any more.
+        # if version < 3:
+        #     self._spot_recovery = None
```

Comment:
I was proposing we keep this uncommented in the history. Any downsides?

Reply:
Please see: https://github.com/skypilot-org/skypilot/pull/3419#discussion_r1584103885 : )

Mostly to keep the codes executed are actually needed. : )